modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C

```diff
@@ -343,26 +343,36 @@ PolygonConcentricCircleMeshGeneratorBase::PolygonConcentricCircleMeshGeneratorBa
         _ring_block_ids.size() !=
             (_ring_intervals.size() + (unsigned int)(num_innermost_ring_layers != 1)))
     {
-      _console << "quad_center_elements is : " << (_quad_center_elements ? "true" : "false")
-               << std::endl;
-      _console << "ring_block_ids size is : " << _ring_block_ids.size() << std::endl;
-      _console << "ring_intervals size is : " << _ring_intervals.size() << std::endl;
-      _console << "first element of ring_intervals is : " << num_innermost_ring_layers << std::endl;
+      // Create an ostringstream for the debug information
+      std::ostringstream debug_info;
+      debug_info << "quad_center_elements is : " << (_quad_center_elements ? "true" : "false")
+                 << std::endl;
+      debug_info << "ring_block_ids size is : " << _ring_block_ids.size() << std::endl;
+      debug_info << "ring_intervals size is : " << _ring_intervals.size() << std::endl;
+      debug_info << "first element of ring_intervals is : " << num_innermost_ring_layers
+                 << std::endl;
+
+      // error message
       if (!_quad_center_elements)
+      {
         paramError("ring_block_ids",
                    "This parameter must have the appropriate size if it is provided: "
                    "If the first element of 'ring_intervals' is more than one, "
                    "the size of 'ring_block_ids' must be "
                    "equal to the size of 'ring_intervals' + 1. Else, equal to the size of "
-                   "'ring_intervals'");
+                   "'ring_intervals'",
+                   debug_info.str());
```

Comment:
Do we need a newline in between here to make sure the output doesn't look like `'ring_intervals'quad_center_elements`?

Reply:
I think that wont hurt at all. I will add it at the beginning of the debug string.